### PR TITLE
fix(overflow-menu): use correct tokens for item icons

### LINKED ISSUE #2051: Overflow menu item icons have incorrect color tokens
from https://github.com/IBM/carbon-components-react/issues/1985

> icons should be color $icon-02 by default and $icon-01 on hover

### PR DESCRIPTION
**Pull request by emyarod**
Closes #2051

This PR adds style rules to keep menu item button icon colors consistent with menu item text colors

Depends on #2053

### CODE REVIEW

**Comment by asudoh** [comment]
Removing "blocked" label as #2053 has been merged.

**Review by asudoh — APPROVED** [review]
LGTM :+1: - Thanks @emyarod!

**Comment by asudoh** [comment]
@emyarod Could you resolve the merge conflict? Thanks!

**Comment by emyarod** [comment]
merge conflict has been resolved @asudoh

File: src/components/overflow-menu/_overflow-menu.scss [hunk 1]
```diff
@@ -329,8 +329,16 @@
...
 
+    svg {
+      fill: $icon-02;
+    }
+
     &:hover {
       color: $text-01;
+
+      svg {
+        fill: $text-01;
```
Comment (alisonjoseph) [inline src/components/overflow-menu/_overflow-menu.scss]: Should this be `$icon-01` ?

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
I think the hover token should be $icon-01 based on Anna's request in the issue

**Review by joshblack — CHANGES_REQUESTED** [review]
Style nit: can we un-nest the structure?

File: src/components/overflow-menu/_overflow-menu.scss (hunk 1 above)
Comment (emyarod) [reply src/components/overflow-menu/_overflow-menu.scss]: yes should be resolved now @alisonjoseph

**Comment by emyarod** [comment]
should be resolved now @joshblack

**Review by shixiedesign — APPROVED** [review]
Looks good! 👍

**Comment by asudoh** [comment]
Let's merge as all pending comments are addressed. Thanks @emyarod and the reviewers!

**Pull request merged**